tests/unit/src/Internal/LoggerTest.php

```diff
@@ -0,0 +1,300 @@
+			->method( 'log' )
+			->with( WC_Log_Levels::INFO, 'Test Log Warning' );
+		$this->logger->info( 'Test Log Warning' );
+	}
+
+
+	/**
+	 * Test debug method of the Logger class.
+	 */
+	public function test_log_level_debug() {
+		$this->logger->expects( $this->once() )
+			->method( 'can_log' )
+			->willReturn( true );
+		$this->mock_wc_logger
+			->expects( $this->once() )
+			->method( 'log' )
+			->with( WC_Log_Levels::DEBUG, 'Test Log Debug' );
+		$this->logger->debug( 'Test Log Debug' );
+	}
+
+	/**
+	 * Test that the can_log method of the Logger class returns true when the mode is dev.
+	 */
+	public function test_can_log_dev_mode() {
+		$this->logger = new Logger(
+			$this->mock_wc_logger,
+			$this->mode,
+			$this->mock_gateway
+		);
+		$this->mode->expects( $this->once() )
+			->method( 'is_dev' )
+			->willReturn( true );
+		$this->assertTrue( $this->logger->can_log() );
+	}
+
+	/**
+	 * Test that the can_log method of the Logger class returns false when the mode throws an exception.
+	 */
+	public function test_can_log_exception() {
+		$this->logger = new Logger(
+			$this->mock_wc_logger,
+			$this->mode,
+			$this->mock_gateway
+		);
+		$this->mode->expects( $this->once() )
+			->method( 'is_dev' )
+			->willThrowException( new Exception() );
+		$this->assertFalse( $this->logger->can_log() );
+	}
+
+	/**
+	 * Test that the can_log method of the Logger class returns false when the gateway option is not set.
+	 */
+	public function test_can_log_no_option() {
+		$this->logger = new Logger(
+			$this->mock_wc_logger,
+			$this->mode,
+			$this->mock_gateway
+		);
+		$this->mode->expects( $this->once() )
+			->method( 'is_dev' )
+			->willReturn( false );
+		$this->mock_gateway->expects( $this->once() )
+			->method( 'get_option' )
+			->with( 'enable_logging' )
+			->willReturn( null );
+		$this->assertFalse( $this->logger->can_log() );
+	}
+
+	/**
+	 * Test that the can_log method of the Logger class returns false when the gateway option is set to 'no'.
+	 */
+	public function test_can_log_disabled() {
+		$this->logger = new Logger(
+			$this->mock_wc_logger,
+			$this->mode,
+			$this->mock_gateway
+		);
+		$this->mode->expects( $this->once() )
+			->method( 'is_dev' )
+			->willReturn( false );
+		$this->mock_gateway->expects( $this->once() )
+			->method( 'get_option' )
+			->with( 'enable_logging' )
+			->willReturn( 'no' );
+		$this->assertFalse( $this->logger->can_log() );
+	}
+
+	/**
+	 * Test that the can_log method of the Logger class returns true when the gateway option is set to 'yes'.
+	 */
+	public function test_can_log_enabled() {
+		$this->logger = new Logger(
+			$this->mock_wc_logger,
+			$this->mode,
+			$this->mock_gateway
+		);
+		$this->mode->expects( $this->once() )
+			->method( 'is_dev' )
+			->willReturn( false );
+		$this->mock_gateway->expects( $this->once() )
+			->method( 'get_option' )
+			->with( 'enable_logging' )
+			->willReturn( 'yes' );
+		$this->assertTrue( $this->logger->can_log() );
+	}
+
```

Comment:
RadoCS: 😄

Reply:
Removed all blank lines

Reply:
There's this one empty line before the closing parenthesis of the class.

Reply:
Done